### PR TITLE
Re-STARR onto 2.13.11-M1 and add JDK 20 (early access) to daily CI matrix

### PR DESCRIPTION
**Pull request by SethTisue**

### CODE REVIEW

File: src/reflect/scala/reflect/internal/util/AbstractFileClassLoader.scala [hunk 1]
```diff
@@ -76,15 +76,20 @@ class AbstractFileClassLoader(val root: AbstractFile, parent: ClassLoader)
...
     case null => null
-    case file => URL.of(new URI(s"memory:${file.path}"), new URLStreamHandler {
+    case file => new URL(null, s"memory:${file.path}", new URLStreamHandler {
       override def openConnection(url: URL): URLConnection = new URLConnection(url) {
         override def connect() = ()
         override def getInputStream = file.input
       }
     })
   }
+
```
Comment (som-snytt) [inline src/reflect/scala/reflect/internal/util/AbstractFileClassLoader.scala:92]: Recently I proposed the compiler should infer EOF on a blank line following 2 or more vertical spacing braces, but I don't know if the rule applies when there is a paren.

Comment (SethTisue) [reply src/reflect/scala/reflect/internal/util/AbstractFileClassLoader.scala:92]: one of the things I like about the Scala 3 indentation-based syntax is that these sorts of questions don't even arise :shrug:  or not nearly as often anyway

Comment (som-snytt) [reply src/reflect/scala/reflect/internal/util/AbstractFileClassLoader.scala:92]: indeed, Scala 3 syntax has only inflamed the provocations on Scala 2.

**Comment by som-snytt** [comment]
So many ex's.
```
!!  2 - run/doc-source-url.scala                  [output differs]
% diff /home/runner/work/scala/scala/test/scaladoc/run/doc-source-url.check /home/runner/work/scala/scala/test/scaladoc/run/doc-source-url-run.log
@@ -1 +1,41 @@
-Done.
+java.net.URISyntaxException: Illegal character in opaque part at index 43: file:test/scaladoc/resources/doc-source-url||.scala||test/scaladoc/resources/doc-source-url.scala||13
```
Might have to change the option to `-doc-source-uri`.

As a prefix of `urinary`, it recalls the childhood joke about Klingons circling Uranus. URL must mean urinal?

**Comment by SethTisue** [comment]
@som-snytt I used to work for an Uri, short for Uriel

just 3 failures now!

```
!!    1 - run/reflection-magicsymbols-invoke.scala  [output differs]
!!    2 - run/t6989                                 [output differs]
!!    3 - run/t9529                                 [output differs]
```

**Comment by som-snytt** [comment]
My brain goes `DarRen STARR` every time. Actually writing it out, it looks creepy, like Darth Vader and Death Star, or the AI build bot that became self-aware and went on a rampage through the computer lab, scrawling `DarRen STARR` on the walls in white board marker.

**Comment by som-snytt** [comment]
Right, these are the tests that fail locally for me. I was going to say that test names all start to look the same.

I wish I'd figured out the allocation tests before they started working again.

**Comment by SethTisue** [comment]
I've removed the "DON'T MERGE ME -- temporarily enable GHA on this PR only" commit now that I've seen green runs.

**Pull request merged**